microservices/optimizer/utils/partition/partition_method.py

```diff
@@ -0,0 +1,23 @@
+"""
+    Enum representing different partition methods.
+"""
+
+from enum import Enum
+from functools import partial
+from .even_subset_split import even_split
+
+
+class PartitionMethod(Enum):
+    """
+    Enum representing different partition methods.
+    """
+
+    EVEN_SPLIT = partial(even_split)
```

Comment:
What does `partial(even_split)` do here? I've never seen this kind of operation on a Python enum
(Not a request for a change, I'm just curious)

Reply:
It allows the function to be stored as a partial function to be used for function application later.